### PR TITLE
Grakn.Concept.Attribute first implementation

### PR DESCRIPTION
**Pull request by alexdeleon**
This is just the minimal implementation to fetch an attribute's value.

The equivalent of: https://github.com/graknlabs/grakn/blob/master/client-nodejs/src/service/Session/concept/Methods.js#L25

### CODE REVIEW

**Comment by kpanic** [comment]
hello @alexdeleon 
This is the dialyzer report:

```elixir
lib/grakn/protocol.ex:48:pattern_match
The pattern
{:error, _reason}

can never match the type
{:ok, Grakn.Transaction.t()}
________________________________________________________________________________
lib/grakn/session.ex:5:invalid_contract
Invalid type specification for function.

Function:
Grakn.Session.new/1

Success typing:
@spec new(binary()) ::
  {:error, _}
  | {:ok,
     %GRPC.Channel{
       :adapter => atom(),
       :adapter_payload => _,
       :cred => %GRPC.Credential{:ssl => [any()]},
       :host => binary(),
       :port => non_neg_integer(),
       :scheme => binary()
     }}
________________________________________________________________________________
lib/grakn/session.ex:50:invalid_contract
Invalid type specification for function.

Function:
Grakn.Session.close/1

Success typing:
@spec close(%GRPC.Client.Stream{
  :canceled => boolean(),
  :channel => %GRPC.Channel{
    :adapter => atom(),
    :adapter_payload => _,
    :cred => %GRPC.Credential{:ssl => [any()]},
    :host => binary(),
    :port => non_neg_integer(),
    :scheme => binary()
  },
  :marshal => (%{:__struct__ => atom(), atom() => _} -> binary()),
  :path => binary(),
  :payload => _,
  :req_stream => boolean(),
  :res_stream => boolean(),
  :unmarshal => (binary() -> %{:__struct__ => atom(), atom() => _})
}) :: :ok
```

I will have a look into the code later on, thanks!

**Comment by kpanic** [comment]
I am trying to address dialyzer reports in #4

**Comment by kpanic** [comment]
Sorry for the conflicts :blush:
Please activate travis in the main repo

File: lib/grakn/concept/action.ex [hunk 1]
```diff
@@ -0,0 +1,25 @@
+defmodule Grakn.Concept.Action do
+  @moduledoc """
+  `ConceptAction` should only be used internally by grakn_elixir to fetch
+  concept information laizily.
```
Comment (kpanic) [inline lib/grakn/concept/action.ex]: lazily

File: lib/grakn/concept/action.ex [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  Sub modules in `Grakn.Concept` should be the only place where these requests are used
+  """
+
+  @type action :: :attribute_value
+  @type t :: %__MODULE__{action: action()}
```
Comment (kpanic) [inline lib/grakn/concept/action.ex]: `:name` instead of action? since the module name is already `Action`?

File: test/grakn_test.exs [hunk 3]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
   setup do
-    {:ok, conn} = Grakn.start_link hostname: "localhost"
+    {:ok, conn} = Grakn.start_link(hostname: "localhost")
+    conn |> Grakn.command(Grakn.Command.delete_keyspace(@keyspace))
     {:ok, conn: conn}
   end
 
   describe "query" do
-
     test "returs a non-empty concept iterator", context do
```
Comment (kpanic) [inline test/grakn_test.exs]: `returns`

File: test/grakn_test.exs [hunk 4]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
-      Grakn.transaction(context[:conn], fn conn ->
-        {:ok, iterator} = Grakn.query(conn, Grakn.Query.graql("match $x; get;"))
-        assert not Enum.empty?(iterator)
-      end)
+      Grakn.transaction(
```
Comment (kpanic) [inline test/grakn_test.exs]: do we care to assert the result of `Grakn.transaction`?

File: test/grakn_test.exs [hunk 5]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
-        Grakn.query!(conn, Grakn.Query.graql("define person sub entity, has name, has age;"))
-      end, type: Grakn.Transaction.Type.write())
-      {:ok, _} = Grakn.transaction(context[:conn], fn conn ->
-        Grakn.query(conn, Grakn.Query.graql("insert $x isa person;"))
-      end, type: Grakn.Transaction.Type.write())
-      {:ok, _} = Grakn.transaction(context[:conn], fn conn ->
-        Grakn.query!(conn, Grakn.Query.graql("match $x isa person; insert $x has name \"test\";"))
-        Grakn.query!(conn, Grakn.Query.graql("match $x isa person; insert $x has age 100;"))
-      end, type: Grakn.Transaction.Type.write())
+      {:ok, _} =
```
Comment (kpanic) [inline test/grakn_test.exs]: missing `assert`

File: test/grakn_test.exs [hunk 6]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
+          fn conn ->
+            Grakn.query!(conn, Grakn.Query.graql("define name sub attribute, datatype string;"))
+            Grakn.query!(conn, Grakn.Query.graql("define age sub attribute, datatype long;"))
+            Grakn.query!(conn, Grakn.Query.graql("define person sub entity, has name, has age;"))
+          end,
+          keyspace: @keyspace,
+          type: Grakn.Transaction.Type.write()
+        )
+
+      {:ok, _} =
```
Comment (kpanic) [inline test/grakn_test.exs]: `assert`

File: test/grakn_test.exs [hunk 7]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
+        Grakn.transaction(
+          context[:conn],
+          fn conn ->
+            Grakn.query(conn, Grakn.Query.graql("insert $x isa person;"))
+          end,
+          keyspace: @keyspace,
+          type: Grakn.Transaction.Type.write()
+        )
+
+      {:ok, _} =
```
Comment (kpanic) [inline test/grakn_test.exs]: `assert`

File: test/grakn_test.exs [hunk 8]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
+          context[:conn],
+          fn conn ->
+            Grakn.query!(conn, Grakn.Query.graql("define name sub attribute, datatype string;"))
+            Grakn.query!(conn, Grakn.Query.graql("define city sub entity, has name;"))
+          end,
+          keyspace: @keyspace,
+          type: Grakn.Transaction.Type.write()
+        )
+
+      {:error, _} =
```
Comment (kpanic) [inline test/grakn_test.exs]: `assert`. Sorry for repetition, let's use `assert`(s) also in the rest of the file

File: test/grakn_test.exs [hunk 9]
```diff
@@ -2,61 +2,147 @@ defmodule GraknTest do
...
     test "incorrect schema", context do
-      {:ok, _} = Grakn.transaction(context[:conn], fn conn ->
-        Grakn.query!(conn, Grakn.Query.graql("define name sub attribute, datatype string;"))
-        Grakn.query!(conn, Grakn.Query.graql("define city sub entity, has name;"))
-      end, type: Grakn.Transaction.Type.write())
-      {:error, _} = Grakn.transaction(context[:conn], fn conn ->
-        {:error, reason} = Grakn.query(conn, Grakn.Query.graql("insert $x isa city; $x has city_name \"wrong\";"))
-        Grakn.rollback(conn, reason)
-      end, type: Grakn.Transaction.Type.write())
+      {:ok, _} =
```
Comment (kpanic) [inline test/grakn_test.exs]: `assert`

File: lib/grakn/transaction.ex [hunk 10]
```diff
@@ -8,36 +11,36 @@ defmodule Grakn.Transaction do
...
-  @spec new(%GRPC.Channel{}) ::
-          {:ok, {%GRPC.Client.Stream{}, []} | {:error, map()} | {:ok, map(), map()}}
+  @spec new(GRPC.Channel.t()) :: {:ok, t()} | {:error, any()}
   def new(channel) do
     req_stream =
       channel
       |> Session.SessionService.Stub.transaction()
 
-    {:ok, {req_stream, []}}
+    with %GRPC.Client.Stream{} <- req_stream do
```
Comment (kpanic) [inline lib/grakn/transaction.ex]: :rocket:

**Comment by alexdeleon** [comment]
Comments addressed @kpanic, merging.

**Pull request merged**